linux_pipeline/Jenkinsfile_kernel_bisect_commit_validator

```diff
@@ -1,435 +1,669 @@
-            name: 'PERF_TEST_THREADS_UDP',
-            description: '[PERF] [UDP] TEST_THREADS.'],
+            choices: """SYNTHETIC
+                    SRIOV""",
+            name: 'NTTTCP_TCP_TYPE',
+            description: 'Azure Performance NETWORK NTTTCP TCP: SYNTHETIC or SRIOV type.'],
         [$class: 'ChoiceParameterDefinition',
-            choices: '4\n1024',
-            name: 'PERF_IO_SIZE',
-            description: '[PERF] [FIO] IO_SIZE.'],
+            choices: """ALL
+                    1\n2\n4\n8\n16\n32\n64\n128\n256\n512\n1024\n2048\n4096\n6144\n8192\n10240""",
+            name: 'NTTTCP_TCP_CONNECTIONS',
+            description: 'Azure Performance NETWORK NTTTCP TCP number of connections.'],
         [$class: 'ChoiceParameterDefinition',
-            choices: '1\n2\n4\n8\n16\n32\n64\n128\n256',
-            name: 'PERF_Q_DEPTH',
-            description: '[PERF] [FIO] Queue depth.'],
+            choices: """SYNTHETIC
+                    SRIOV""",
+            name: 'NTTTCP_UDP_TYPE',
+            description: 'Azure Performance NETWORK NTTTCP UDP: SYNTHETIC or SRIOV type.'],
         [$class: 'ChoiceParameterDefinition',
-            choices: 'read\nrandread\nwrite\nrandwrite',
-            name: 'PERF_IO_MODE',
-            description: '[PERF] [FIO] IO_MODE']
+            choices: """ALL
+                    2\n4\n8\n16\n32\n64\n128\n256\n512\n1024""",
+            name: 'NTTTCP_UDP_CONNECTIONS',
+            description: 'Azure Performance NETWORK NTTTCP UDP number of connections.'],
+        [$class: 'StringParameterDefinition',
+            name: 'CUSTOM_BUILD_TAG',
+            defaultValue: "",
+            description: 'Custom build tag for artifacts. If left undefined, BUILD_NUMBER will be used']
         ]
     ]
 ])
 
-perfXml = ["TCP": "TCP_SRIOV_v4.xml", "UDP":"UDP_SRIOV_v4.xml", "FIO": "FIO_RAID0_Pipeline.xml"]
+// ****************
+// PIPELINE HELPERS
+// ****************
 
-env.BUILD_PATH = '/mnt/tmp/bisect-kernel-build-folder'
-env.KERNEL_ARTIFACTS_PATH = 'bisect-kernel-artifacts'
-env.KERNEL_CONFIG = 'Microsoft/config-azure'
-env.BUILD_NAME = 'b'
-env.INSTALL_DEPS = 'True'
-env.USE_KERNEL_FOLDER_PREFIX = 'True'
-env.THREAD_NUMBER = 'x3'
-env.AZURE_MAX_RETRIES = '10'
+def RunPowershellCommand(psCmd) {
+    bat "powershell.exe -NonInteractive -ExecutionPolicy Bypass -Command \"[Console]::OutputEncoding=[System.Text.Encoding]::UTF8;$psCmd;EXIT \$global:LastExitCode\""
+}
 
-isPerf = BISECT_TYPE.equals('Performance')
-isBootTest = BISECT_TYPE.equals('Boot')
-isFunc = BISECT_TYPE.equals('Functional')
+def getVhdLocation(basePath, distroVersion) {
+    def distroFamily = distroVersion.split('_')[0]
+    return "${basePath}\\" + distroFamily + "\\" + distroVersion + "\\" + distroVersion + ".vhdx"
+}
+
+def prepareEnv(branch, remote) {
+    cleanWs()
+    git branch: branch, url: remote
+}
 
-perfServer = GetPerfServer(PERF_TEST)
-perfTestType = GetPerfTestType(PERF_TEST)
+def unstashKernel(kernelStash) {
```

Comment:
i think is a better idea to add another parameter to specify what to keep, and delete everything else

Reply:
will keep in mind and I will do it when I refactor the developer/ bisect slave pipelines in one.